pkg/apis/controlplane/v1beta2/types.go

```diff
@@ -218,9 +218,10 @@ type Service struct {
 	// field defaults to TCP.
 	// +optional
 	Protocol *Protocol `json:"protocol,omitempty" protobuf:"bytes,1,opt,name=protocol"`
-	// The port name or number on the given protocol. If not specified, this matches all port numbers.
+	// A PortMask is a representation of a single port name/number or a range of ports
+	// from base port with a bitmask applied.
 	// +optional
-	Port *intstr.IntOrString `json:"port,omitempty" protobuf:"bytes,2,opt,name=port"`
+	PortMask *PortMask `json:"portMask,omitempty" protobuf:"bytes,3,opt,name=portMask"`
```

Comment:
this is a very good point.. especially if in future we have multiple data planes to support. adding a new endPort field to types here is sound because that does not break backwards compatibility as an empty port and endPort with value will not be allowed by controller.

Reply:
On the compatibility front... I assumed the case @tnqn mentioned would be covered by `Convert_controlplane_Service_To_v1beta1_Service`? Nevertheless I agree for the controlplane implementation to be dataplane agnostic it makes sense to change the portMask field to endPort here.

Reply:
I didn't follow this PR closely, but Quan's points seem reasonable and it makes sense to implement the conversion from port range to list of port + mask in the agent, as this is really OpenFlow-specific. I thought we had agreed a while back to do it in the agent, when we discussed whether this should be implemented in Antrea or in libOpenflow?